tests/programs/custom/zalrsc/zalrsc.c

```diff
@@ -301,11 +316,43 @@ int test_SCW_misaligned_access(void)
 
   //(Execute exception handler)
 
-  if (num_misaligned_SCW_trapped != 1) {
+  if (misaligned_SCW_trapped != 1) {
     printf("Expected SCW to trap due to misaligned address. However, it did not.\n");
     return 1;
   }
 
-  num_misaligned_SCW_trapped = 0;
+  misaligned_SCW_trapped = 0;
+  return 0;
+}
+
+int test_zalrsc_to_non_atomic_region(uint8_t is_scw)
+{
+  printf("\n\nTest using LR.W on non atomic memory region.\n\n");
+
+  volatile uint32_t nonatomic_region_addr = 0xFFFFFFFE;
+  volatile uint32_t is_failure = 0;
+  volatile uint32_t random = 0;
+  trap_handler_beh = NONATOMIC_REGION;
+
+  if (is_scw) {
+    __asm__ volatile("sc.w %[is_failure], %[random], (%[nonatomic_region_addr])"
+    : [is_failure]"=r"(is_failure)
+    : [random] "r"(random), [nonatomic_region_addr]"r"(nonatomic_region_addr));
+
+  } else {
+    __asm__ volatile("lr.w %[random], (%[nonatomic_region_addr])"
+    : [random]"=r"(random)
+    : [nonatomic_region_addr]"r"(nonatomic_region_addr));
+  }
+
+  //(Execute exception handler)
+
+  if (!zalrsc_nonatomic_region_trapped) {
```

Comment:
`zalrsc_nonatomic_region_trapped` is not initialized? So if the trap didn't occur, could the test still pass?

Reply:
True! FIXED